trio/_core/_io_epoll.py

```diff
@@ -8,6 +9,8 @@
 from ._io_common import wake_all
 from ._wakeup_socketpair import WakeupSocketpair
 
+assert sys.platform == "linux"
```

Comment:
Maybe we should put these inside an `if TYPE_CHECKING:`? Both to make it clear that the purpose of this `assert` is  for the benefit of type-checkers, and because we've generally avoided doing actual `sys.platform` checks at runtime.

Reply:
If I put the asserts in `if TYPE_CHECKING:`, mypy no longer recognizes them!

Reply:
Which is bad. We've talked about this with @oremanj in chat: https://gitter.im/python-trio/general?at=5ef98c54e0e5673398df291c